### ISSUE TITLE
BigQuery: `MERGE` statements without source alias are unparsable

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3270]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In BigQuery a `MERGE` statement using a sub-query as `SOURCE` does not (technically) require an alias. However this https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_ansi.py#L2357-L2361 makes these statements unparsable.

For example

```sql
MERGE dataset.NewArrivals
USING (SELECT * FROM dataset.NewArrivals WHERE warehouse <> 'warehouse #2')
ON FALSE
WHEN MATCHED THEN
  DELETE
```

is a valid BigQuery SQL. (Whether or not it is a meaningful query to execute, is beyond scope of this issue 😂 )

### Expected Behaviour

SQLFluff can parse `MERGE` statement like in the example.

Additional: Probably this should be somehow integrated with SQLFluff rule sets that deal with table aliases? `L011`, `L020`, `L025`, etc...

### Observed Behaviour

```
Line 1, Position 1: Found unparsable section: 'MERGE dataset.NewArrivals
USING (SELECT ...'
```

### How to reproduce

https://online.sqlfluff.com/fluffed?sql=H4sIADLxdGIC__N1DXJ3VUhJLEksTi3R80stdywqyixLzCnm5QoN9vRzV9AIdvVxdQ5R0FJwC_L3xaZSIdzDNchVoTyxKDUjv7Q4VcHGTkEdwVM2Utfk5fL3U3Bz9Al25eUCqvZT8HUMcfZwdVEIAXJ4uRQUXICWhAAlAd1JCSCPAAAA&dialect=bigquery

### Dialect

bigquery

### Version

0.13.0 -> webversion

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
